astropy/units/tests/test_units.py

```diff
@@ -371,6 +371,7 @@ def test_compose_cgs_to_si(unit):
     si = unit.to_system(u.si)
     assert [x.is_equivalent(unit) for x in si]
     assert si[0] == unit.si
+    assert np.isclose(si[0].scale, unit.decompose(bases=u.si.bases).scale)
```

Comment:
I added these to verify my wishlist, point 1. Obviously not exhaustive, but better than not testing at all...